File: array_merge/array_merge.java
import java.lang.reflect.Array;
import java.util.Arrays;

public class array_merge {
    // ATOMIC_FUNCTION:  array_merge(T a, T b)
    // TAGS: list
    // LANG:  java
    // FUNCTION: array_merge(T a, T b)
    public static <T> T array_merge(T a, T b) {
        if (!a.getClass().isArray() || !b.getClass().isArray()) {
            throw new IllegalArgumentException();
        }

        Class<?> resCompType;
        Class<?> aCompType = a.getClass().getComponentType();
        Class<?> bCompType = b.getClass().getComponentType();

        if (aCompType.isAssignableFrom(bCompType)) {
            resCompType = aCompType;
        } else if (bCompType.isAssignableFrom(aCompType)) {
            resCompType = bCompType;
        } else {
            throw new IllegalArgumentException();
        }

        int aLen = Array.getLength(a);
        int bLen = Array.getLength(b);

        @SuppressWarnings("unchecked")
        T result = (T) Array.newInstance(resCompType, aLen + bLen);
        System.arraycopy(a, 0, result, 0, aLen);
        System.arraycopy(b, 0, result, aLen, bLen);

        return result;
    }
    public static void main(String args[]) {
        String[] start_val_1 = {"hello","world"};
        String[] start_val_2 = {"atomic"};
        String[] expected_end_val={"hello","world","atomic"};
        array_merge a= new array_merge();

        String[] end_val = a.array_merge(start_val_1,start_val_2);

        // TEST
        if(Arrays.equals(end_val,expected_end_val)) {
            System.out.println("PASS");
        } else {
            System.out.println("FAIL");
        }
    }

}

File: array_merge/array_merge.py
# ATOMIC_FUNCTION: array_merge( first_array , second_array )
# TAGS: list
# LANG:  python
# FUNCTION: array_merge( first_array , second_array )
def array_merge( first_array , second_array ):
	if isinstance( first_array , list ) and isinstance( second_array , list ):
		return first_array + second_array
	elif isinstance( first_array , dict ) and isinstance( second_array , dict ):
		return dict( list( first_array.items() ) + list( second_array.items() ) )
	elif isinstance( first_array , set ) and isinstance( second_array , set ):
		return first_array.union( second_array )
	return False

start_val_1 = ["hello","world"]
start_val_2 = ["atomic"]
expected_end_val = ["hello","world","atomic"]

end_val = array_merge(start_val_1,start_val_2)

# TEST
if(end_val == expected_end_val):
    print("PASS")
else:
    print("FAIL")







